exploration/text-vs-code.md

```diff
@@ -101,32 +144,62 @@ Which whitespace characters are displayed at runtime should not be surprising.
+
+—<a href="https://www.infoq.com/articles/API-Design-Joshua-Bloch/">Joshua Bloch, 2008</a>, author of <i>Effective Java</i>, etc.
+</blockquote>
+<blockquote>
+The Pit of Success: in stark contrast to a summit, a
+peak, or a journey across a desert to find victory through many trials and
+surprises, we want our customers to simply fall into winning practices by using
+our platform and frameworks. To
+the extent that we make it easy to get into trouble we fail.
+          
+—Rico Mariani, MS Research MindSwap Oct 2003. (<a href="https://learn.microsoft.com/en-us/archive/blogs/brada/the-pit-of-success">restated by Brad Adams</a>, MS CLR and .Net team cofounder)
+</blockquote>
+</details>
 
 Developers and translators should be able to read and write the syntax easily in a text editor.
 
 Translators (and their tools) are not software engineers, so we want our syntax
-to be as simple, robust, and non-fussy as possible.
-Multiple levels of complex nesting should be avoided,
-along with any constructs that require an excessive
-level of precision on the part of non-technical authors.
+to be as simple and robust as possible.
+
+Nesting level is not a requirement.
+People are not parsers, and don't care about the nesting level as a primary concern when reading a message.
+What matters to them is their ability to recognize where a message's pattern starts and where it ends.
+In the following example, localizable text is easily recognizable (especially with syntax highlighting),
+even if it occurs 3 level deep.
+
+```java
+print "{{{This is translatable}}}"
+if (foo) print "{{{This is translatable}}}" else print "{{{This is NOT translatable}}}"
+if (foo) if (bar) switch (baz) case 1: print "{{{This is translatable, deep}}}" break; default: print "{{{This is NOT translatable, deep}}}"
+```
 
 As MessageFormat 2 will be at best a secondary language to all its authors and editors,
 it should conform to user expectations and require as little learning as possible.
 
-The syntax should avoid footguns,
-in particular as it's passed through various tools during formatting.
+The syntax should avoid footguns, in particular as it's passed through various tools during formatting or stored existing file formats, databases, etc.
+Very importantly in this regard,
+we should minimize the range of characters that need to be escaped in patterns.
+
 
 ASCII-compatible syntax. While support for non-ASCII characters for variable names,
 values, literals, options, and the like are important, the syntax itself should
 be restricted to ASCII characters. This allows the message to be parsed
 visually by humans even when embedded in a syntax that requires escaping.
 
-Whitespace is forgiving.
-We _require_ the minimum amount of whitespace and allow
-authors to format or change unimportant whitespace as much as they want.
```

Comment:
Why did you remove this? It seems like a good observation about all of our designs.